packages/util/src/functions/userAgent.ts

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable n/prefer-global/process */
+/* eslint-disable no-restricted-globals */
+
+/**
+ * Resolves the user agent string for the current environment.
+ */
+export function getUserAgent(): string | null {
+	// Most (if not all) edge environments will have `process` defined. Withing a web browser we'll extract it using `navigator.userAgent`.
+	return typeof process === 'object'
+		? process.release?.name === 'node'
+			? `Node.js/${process.version}`
+			: ''
+		: // @ts-expect-error web env
+		typeof window === 'object'
+		? // @ts-expect-error web env
+		  window.navigator.userAgent
+		: null;
```

Comment:
This should also handle deno and bun for the UA

```suggestion
	return typeof globalThis.process === 'object'
		? globalThis.process.release?.name === 'node'
			? `Node.js/${globalThis.process.version}`
			: ''
		: // @ts-expect-error web env
		typeof globalThis.window === 'object'
		? // @ts-expect-error web env
		  globalThis.window.navigator.userAgent
		: null;
```